### PR TITLE
[Caffe2] Enable channel shuffle op on IDEEP device with openomp optimization

### PR DESCRIPTION
**Pull request by gujinghui**
1. Enable channel shuffle op on IDEEP device
2. Optimize channel shuffle by openmp
3. Add channel shuffle test case

### CODE REVIEW

File: caffe2/ideep/CMakeLists.txt [hunk 1]
```diff
@@ -9,7 +9,7 @@ if(USE_MKL AND USE_IDEEP AND CAFFE2_COMPILER_SUPPORTS_AVX2_EXTENSIONS)
 
   add_library(Caffe2_ideep_operators OBJECT ${avx2_srcs})
   add_dependencies(Caffe2_ideep_operators Caffe_PROTO Caffe2_PROTO)
-  set_target_properties(Caffe2_ideep_operators PROPERTIES COMPILE_FLAGS "-mavx2")
+  set_target_properties(Caffe2_ideep_operators PROPERTIES COMPILE_FLAGS "-mavx2 -fopenmp")
```
Comment (yinghai) [inline caffe2/ideep/CMakeLists.txt]: BTW, could you remind me why do we need `-mavx2` in IDEEP?

**Comment by yinghai** [comment]
I have some comments on https://github.com/intel/ideep/commit/55a293c4f71efebbef389cf00e1637fc36ffa621 too.

**Comment by gujinghui** [comment]
@yinghai  replied your comments on intel/ideep@55a293c

File: caffe2/ideep/CMakeLists.txt (hunk 1 above)
Comment (gujinghui) [reply caffe2/ideep/CMakeLists.txt]: yes. if have to impl optimized op required by caffe2 in ideep in time in future...

Comment (yinghai) [reply caffe2/ideep/CMakeLists.txt]: Hmm, as far as I understand mkl-dnn will jit code based on the cpuinfo. Where do we need avx2 for ideep? Do you have an example?

Comment (gujinghui) [reply caffe2/ideep/CMakeLists.txt]: for example, channel_shuffle.
If mkldnn has no plan on it, avx2 intrinsic is an option to impl it in ideep for non-public format...

**Comment by gujinghui** [comment]
@yinghai  pls help review and merge.

File: caffe2/ideep/CMakeLists.txt (hunk 1 above)
Comment (yinghai) [reply caffe2/ideep/CMakeLists.txt]: I think `mkldnn` does have a plan for it.

Comment (gujinghui) [reply caffe2/ideep/CMakeLists.txt]: there should be a gap between a plan and code ready...
anyway, avx2 is optional. you can remove it for now.

**Comment by yinghai** [comment]
@gujinghui Is this still relevant?

**Comment by gujinghui** [comment]
@yinghai 
Not needed now.
Closed.

**Pull request closed without merging**